app/models/concerns/findings/issues.rb

```diff
@@ -9,6 +9,54 @@ module Findings::Issues
     accepts_nested_attributes_for :issues, allow_destroy: true, reject_if: :all_blank
   end
 
+  def issues_amount
+    issues.sum &:amount
+  end
+
+  def get_amount_by_impact
+    amount = issues_amount
+
+    amount_by_impact.reverse_each.to_h.detect { |id, value| amount >= value }
+  end
+
+  def impact_risk_text
+    I18n.t "impact_risk_types.#{Finding::IMPACT_RISKS.invert[get_amount_by_impact.first]}"
+  end
+
+  def impact_risk_value
+    get_amount_by_impact&.first
+  end
+
+  def probability_risk_previous
+    if weakness_template_id
+      quantity = 1
+
+      4.times do
+        review = review&.previous
+
+        if review && (previous_weakness_by_templat? review)
```

Comment:
```suggestion
        if review && previous_weakness_by_template?(review)
```
Y fijate que estaba mal escrito, PROBALO, acordate de esa parte. Con los ojos es jodido a veces detectar esos casos.